### PR TITLE
Add googletest for testing & add filesystem test suite

### PR DESCRIPTION
**Pull request by feliwir**
- [x] Add option for enabling testing (via GoogleTest and CTest)
- [x] Integrate into GitHub workflows
- [x] Add different kind of tests to show usage (for filesystem & bigarchives)

### CODE REVIEW

File: src/w3d/lib/cpudetect.cpp [hunk 1]
```diff
@@ -273,6 +273,11 @@ void Get_OS_Info(OSInfoStruct &os_info,
             os_info.VersionString = "UNKNOWN";
             break;
     }
+#else
+    memset(&os_info, 0, sizeof(os_info));
```
Comment (duncanspumpkin) [inline src/w3d/lib/cpudetect.cpp]: This is c++ can't you just do `os_info = OSInfoStruct {};`

Comment (feliwir) [reply src/w3d/lib/cpudetect.cpp]: I copied this from 5 lines above. However i can change it in both places

Comment (duncanspumpkin) [reply src/w3d/lib/cpudetect.cpp]: Oh yes so you did. just leave as is.

Comment (feliwir) [reply src/w3d/lib/cpudetect.cpp]: I still fixed this in my last commit

**Comment by duncanspumpkin** [comment]
Shouldn't these tests be purely for standalone and therefore it can be expected that std::filesystem will be available.

**Comment by feliwir** [comment]
> Shouldn't these tests be purely for standalone and therefore it can be expected that std::filesystem will be available.

Right now these tests are a seperate option and work for both standalone & not standalone - that's why i check for the option

**Comment by duncanspumpkin** [comment]
> > Shouldn't these tests be purely for standalone and therefore it can be expected that std::filesystem will be available.
> 
> Right now these tests are a seperate option and work for both standalone & not standalone - that's why i check for the option

I just fear you are going to need lots of #ifdef.

**Comment by feliwir** [comment]
Well the ifdef for BUILD_WITH_STDFS is required in all cases - it can always be disabled, no matter if standalone or not

**Review by duncanspumpkin — APPROVED** [review]
All looks sensible to me but since its new infrastructure @OmniBlade should have final say.

**Review by OmniBlade — CHANGES_REQUESTED** [review]
Just a few files need newline endings, other than that I'm happy to merge it.

**Comment by feliwir** [comment]
> Just a few files need newline endings, other than that I'm happy to merge it.

Done

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**